das/bigcache_storage_service.go

```diff
@@ -0,0 +1,89 @@
+// Copyright 2022, Offchain Labs, Inc.
+// For license information, see https://github.com/nitro/blob/master/LICENSE
+
+package das
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	"github.com/allegro/bigcache"
+	flag "github.com/spf13/pflag"
+
+	"github.com/ethereum/go-ethereum/crypto"
+)
+
+type BigCacheConfig struct {
+	// TODO add other config information like HardMaxCacheSize
+	Expiration time.Duration `koanf:"big-cache-expiration"`
+}
+
+var DefaultBigCacheConfig = BigCacheConfig{
+	Expiration: time.Hour,
+}
+
+func BigCacheConfigAddOptions(prefix string, f *flag.FlagSet) {
+	f.Duration(prefix+".big-cache-expiration", DefaultBigCacheConfig.Expiration, "Big cache expiration")
+}
+
+type BigCacheStorageService struct {
+	baseStorageService StorageService
+	bigCacheConfig     BigCacheConfig
+	bigCache           *bigcache.BigCache
+}
+
+func NewBigCacheStorageService(bigCacheConfig BigCacheConfig, baseStorageService StorageService) (StorageService, error) {
+	bigCache, err := bigcache.NewBigCache(bigcache.DefaultConfig(bigCacheConfig.Expiration))
+	if err != nil {
+		return nil, err
+	}
+	return &BigCacheStorageService{
+		baseStorageService: baseStorageService,
+		bigCacheConfig:     bigCacheConfig,
+		bigCache:           bigCache,
+	}, nil
+}
+
+func (bcs *BigCacheStorageService) GetByHash(ctx context.Context, key []byte) ([]byte, error) {
+	ret, err := bcs.bigCache.Get(string(key))
+	if err != nil {
```

Comment:
I think we need to check that we've actually found the right value here. From the bigcache docs:

> BigCache does not handle collisions. When new item is inserted and it's hash collides with previously stored item, new item overwrites previously stored value.

Reply:
Actually that is already being done inside the library itself https://github.com/allegro/bigcache/blob/master/shard.go#L69

So if there is a collision and if due to that actual key is not present, it returns not found.